backend/eventloop/workspace_event_loop_test.go

```diff
@@ -69,15 +93,234 @@ var _ = Describe("Workspace Event Loop Test", func() {
+			}()
+
+			// Consider test case passed if a new application event loop is not created in 5 seconds.
+			Consistently(tAELF.numberOfEventLoopsCreated, "5s").Should(Equal(originalNumberOfEventLoopsCreated),
+				"the number of event loops shoulnd't change")
+		})
+
+		It("Should unorphan previous GitOpsDeploymentSyncRun event if parent GitOpsDeployment event is passed and new application event loop should be created.", func() {
+
+			By("creating a new GitOpsDeployment resource, that the GitOpsDeploymentSyncRun was missing from the previous step.")
+
+			gitopsDepl := &managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "dummy-deployment",
+					Namespace: apiNamespace.Name,
+					UID:       "D",
+				},
+				Spec: managedgitopsv1alpha1.GitOpsDeploymentSpec{
+					Source: managedgitopsv1alpha1.ApplicationSource{},
+					Type:   managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated,
+				},
+			}
+
+			k8sClientOuter := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(gitopsDepl, apiNamespace, argocdNamespace, kubesystemNamespace).
+				Build()
+
+			// Simulate a GitOpsDeployment modified event
+			msg := eventlooptypes.EventLoopMessage{
+				MessageType: eventlooptypes.ApplicationEventLoopMessageType_Event,
+				Event: &eventlooptypes.EventLoopEvent{
+					EventType: eventlooptypes.DeploymentModified,
+					Request: reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Namespace: apiNamespace.Namespace,
+							Name:      gitopsDepl.Name,
+						},
+					},
+					Client:                  k8sClientOuter,
+					ReqResource:             managedgitopsv1alpha1.GitOpsDeploymentTypeName,
+					AssociatedGitopsDeplUID: string(gitopsDepl.UID),
+					WorkspaceID:             string(apiNamespace.UID),
+				},
+			}
+
+			// Create goroutine and pass event,
+			//because loop in workspaceEventLoopRouter will keep on waiting for new event to be received and test will get stuck here.
+			go func() {
+				inputChannel <- msg
+			}()
+
+			Eventually(func() bool {
+				// We use a function here to check if number is 3
+				return tAELF.numberOfEventLoopsCreated == 3
+			}).Should(BeTrue())
+
+			By("Read the events from the output channel, make sure they are the ones we expect, now that the gitopsdeploymentsyncrun in unorphaned")
+
+			deploymentModifiedMsg := <-tAELF.outputChannel
+
+			// Make sure this is the gitopsdeployment event from above
+			Expect(deploymentModifiedMsg).NotTo(BeNil())
+			Expect(deploymentModifiedMsg.Event).To(Equal(msg.Event))
+			Expect(deploymentModifiedMsg.MessageType).To(Equal(msg.MessageType))
+
+			syncRunModifiedMsg := <-tAELF.outputChannel
 
+			// Make sure this is the gitopsdeploymentsyncrun event from above
+			Expect(syncRunModifiedMsg).NotTo(BeNil())
+			Expect(syncRunModifiedMsg.Event.Request.Name).To(Equal(msgTemp.Event.Request.Name))
+			Expect(deploymentModifiedMsg.MessageType).To(Equal(msgTemp.MessageType))
```

Comment:
Guessing this line (314) is supposed to be `Expect(syncRunModifiedMsg)`? At the moment it's the same as a line above.

Reply:
Yes, I missed to change it after copying from above :).